admiral/envs/components/resources.py

```diff
@@ -136,31 +126,42 @@ def regrow(self, **kwargs):
         self.resources[self.resources >= self.min_value] += self.regrow_rate
         self.resources[self.resources >= self.max_value] = self.max_value
 
-    def render(self, fig=None, **kwargs):
+class GridResourcesActor:
+    def __init__(self, resources=None, agents=None, **kwargs):
+        self.resources = resources
+        self.agents = agents
+
+        from gym.spaces import Box
+        for agent in agents.values():
+            if isinstance(agent, HarvestingAgent):
```

Comment:
Harvest agent must be position agent because harvesting is based on position. In order to avoid weird hierarchy of agents, we leave them unrelated. However, here we should assert that the agents are also position agents.

Reply:
Do this everywhere else where it applies

Reply:
Already done in ResourceState